students/kevin_t/Lesson4/basic_operations.py

```diff
@@ -0,0 +1,105 @@
+"""
+Store customer data from HP Norton in a relational database (sqlite3).
+"""
+import logging
+from peewee import SqliteDatabase, Model, IntegerField, CharField, BooleanField, DecimalField
+
+database = SqliteDatabase('customer.db')
+database.connect()
+database.execute_sql('Pragma foreign_keys = ON;')
+
+class BaseModel(Model):
+    """ Set up database """
+    class Meta:
+        """ Establish database """
+        database = database
+
+class Customer(BaseModel):
+    """
+        This class defines a customer, which maintains the details of
+        a customer's information.
+    """
+
+    customer_id = IntegerField(primary_key=True)
+    first_name = CharField()
+    last_name = CharField()
+    home_address = CharField()
+    phone_number = IntegerField()
+    email_address = CharField()
+    status = BooleanField()
+    credit_limit = DecimalField()
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+def add_customer(customer_id, first_name, last_name, home_address, phone_number,
+                 email_address, status, credit_limit):
+    """ Create a new customer profile """
+    try:
+        with database.transaction():
+            new_customer = Customer.create(customer_id=customer_id, first_name=first_name,
+                                           last_name=last_name, home_address=home_address,
+                                           phone_number=phone_number, email_address=email_address,
+                                           status=status, credit_limit=credit_limit)
+            new_customer.save()
+            LOGGER.info(f'Database add successful {first_name} {last_name}')
+
+    except Exception as error_info:
```

Comment:
Avoid using a _catch all_ exception. The risk with these is that they will catch (and handle) ANY exceptions, expected and unexpected, which might cause new issues to be hidden away, complicating the debug process. Instead, use specific exceptions that you know could occur or remove your _try/except_ block altogether.